build.groovy

```diff
@@ -516,11 +502,43 @@ def finalizeBuildProcess(Map args) {
 			}
 		}
 
-		// save build result
+		// add files processed and set state
+		buildResult.setProperty("filesProcessed", String.valueOf(args.count))
+		buildResult.setState(buildResult.COMPLETE)
+
+		// save updates
 		buildResult.save()
 
+		// store build result properties in BuildReport.json
+		PropertiesRecord buildReportRecord = new PropertiesRecord("DBB.BuildResultProperties")
+		def buildResultProps = buildResult.getPropertyNames()
+		buildResultProps.each { buildResultPropName ->
+			buildReportRecord.addProperty(buildResultPropName, buildResult.getProperty(buildResultPropName))
+		}
+		buildReport.addRecord(buildReportRecord)
 	}
 
+	// create build report html file
+	def htmlOutputFile = new File("${props.buildOutDir}/BuildReport.html")
+	println "** Writing build report to ${htmlOutputFile}"
+	def htmlTemplate = null  // Use default HTML template.
+	def css = null       // Use default theme.
+	def renderScript = null  // Use default rendering.
+	def transformer = HtmlTransformer.getInstance()
+	transformer.transform(jsonOutputFile, htmlTemplate, css, renderScript, htmlOutputFile, buildReportEncoding)
+	
+	// save json file
```

Comment:
Another minor issue.  Can you create and save the data file before the html file?  I know it does not matter functionally but since its passed into the html traqnsformer, logically it should be created and saved before the transformer is called.